hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -169,10 +170,19 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
 
     TableOperations ops = newTableOps(identifier);
     TableMetadata lastMetadata;
-    if (purge && ops.current() != null) {
-      lastMetadata = ops.current();
-    } else {
-      lastMetadata = null;
+    try {
+      if (purge && ops.current() != null) {
+        lastMetadata = ops.current();
+      } else {
+        lastMetadata = null;
+      }
+    } catch (Exception e) {
```

Comment:
Can directly just catch `NotFoundException`? Instead of catching all the exceptions and throwing again if it is not a `NotFoundException`?

Reply:
Yes, I'm sorry, my compiler had a problem before, it has been modified